corehq/apps/accounting/tasks.py

```diff
@@ -698,45 +698,56 @@ def assign_explicit_community_subscription(domain_name, start_date, method, acco
 def run_downgrade_process():
     today = datetime.date.today()
 
-    for domain, oldest_unpaid_invoice, total in _get_domains_with_subscription_invoices_over_threshold(today):
+    for domain, oldest_unpaid_invoice, total in get_domains_with_subscription_invoices_over_threshold(today):
         current_subscription = Subscription.get_active_subscription_by_domain(domain)
-        if _is_subscription_eligible_for_downgrade_process(current_subscription):
+        if is_subscription_eligible_for_downgrade_process(current_subscription):
             _apply_downgrade_process(oldest_unpaid_invoice, total, today, current_subscription)
 
-    for oldest_unpaid_invoice, total in _get_accounts_with_customer_invoices_over_threshold(today):
+    for oldest_unpaid_invoice, total in get_accounts_with_customer_invoices_over_threshold(today):
         subscription_on_invoice = oldest_unpaid_invoice.subscriptions.first()
-        if _is_subscription_eligible_for_downgrade_process(subscription_on_invoice):
+        if is_subscription_eligible_for_downgrade_process(subscription_on_invoice):
             _apply_downgrade_process(oldest_unpaid_invoice, total, today)
 
 
-def _get_domains_with_subscription_invoices_over_threshold(today):
-    unpaid_saas_invoices = Invoice.objects.filter(
```

Comment:
So earlier `unpaid_saas_invoices` was a queryset and was being re-used to get `total_overdue_to_date`. And now since these are in separate methods its being initialised each time for a domain. I guess better to still keep that in a way it can be reused,